src/lib/core/bidi/directionality.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  EventEmitter,
+  Injectable,
+  Optional,
+  SkipSelf
+} from '@angular/core';
+
+export type Direction = 'ltr' | 'rtl';
+
+/**
+ * The directionality (LTR / RTL) context for the application (or a subtree of it).
+ * Exposes the current direction and a stream of direction changes.
+ */
+@Injectable()
+export class Directionality {
+  value: Direction = 'ltr';
+  public change = new EventEmitter<void>();
+
+  constructor() {
+    if (typeof document) {
```

Comment:
`typeof document === 'object' && !!document`.

We have an `isBrowser` property on `Platform` now for things like this, but we can adjust that later (since it involves bringing in `PlatformModule`)